core/String.carp

```diff
@@ -169,6 +170,19 @@
 
   (doc contains? "Checks whether the string `s` contains the character `c`.")
   (defn contains? [s c] (> (index-of s c) -1))
+
+  (private  tolower-)
+  (hidden   tolower-)
+  (register tolower- (Fn [Byte] Byte) "tolower")
+  (private  toupper-)
+  (hidden   toupper-)
+  (register toupper- (Fn [Byte] Byte) "toupper")    
```

Comment:
I would argue that this shouldn’t be part of `String`, but may be better suited for the `Byte` module. It could still be `hidden`, just not `private`.

Reply:
Yes, this is the _ugly_ bit, so I tried to hide it as much as possible...

But since it's about converting characters, wouldn't the Char module be a better fit?

Reply:
In a way that would make sense, but I would go by the type here usually. We could make this a function that operates on `CChar`, but that would probably just complicate matters even more.

Reply:
done

Reply:
Thanks!